src/main/java/build/buildfarm/instance/stub/ByteStreamUploader.java

```diff
@@ -393,6 +393,17 @@ public void onMessage(WriteResponse response) {
 
             @Override
             public void onClose(Status status, Metadata trailers) {
+              try {
+                // If upload was completed by someone else or already present, then chunker does
+                // does not close the Filehandle. Do it here to be sure it's always closed.
+                chunker.reset();
+              } catch (IOException e1) {
+                // This exception indicates that closing the underlying input stream failed.
+                // We don't expect this to ever happen, but don't want to swallow the exception
+                // completely.
+                logger.log(Level.WARNING, format("Chunker failed closing data source: %s", e1));
```

Comment:
Change `e1` to `e` here, and take it out of the format - we supply exceptions as the final parameter to `log` calls so that the handlers can deal with the traces as they see fit - this will not print a stack trace.

Reply:
Updated